From 2e38daa617fc44f06687583023a01319d804f71c Mon Sep 17 00:00:00 2001
From: Patrick Delaunay <patrick.delaunay@foss.st.com>
Date: Mon, 20 Jan 2025 18:54:43 +0100
Subject: [PATCH 789/954] fix(stm32mp2): clear low-power context on standby
 exit

Clear the low-power context during the standby exit to avoid its usage
for next wake-up; this patch forces a cold boot for next BL2 execution
if stm32_pm_context_save() is not called, i.e. if a Standby  sequence
is not requested.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 542c75bcd042da3b59c8fc3f04976d9d06c14b2f fix(stm32mp2): clear low-power context on standby exit]
Signed-off-by: Patrick Delaunay <patrick.delaunay@foss.st.com>
Change-Id: I35851c194059732338d1b4837789c08037b025a0
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/430163
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Maxime MERE <maxime.mere@foss.st.com>
---
 plat/st/stm32mp2/stm32mp2_pm.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/plat/st/stm32mp2/stm32mp2_pm.c b/plat/st/stm32mp2/stm32mp2_pm.c
index 69db420563..5d2b64bbd7 100644
--- a/plat/st/stm32mp2/stm32mp2_pm.c
+++ b/plat/st/stm32mp2/stm32mp2_pm.c
@@ -1465,6 +1465,9 @@ int plat_setup_psci_ops(uintptr_t sec_entrypoint,
 void stm32_pm_context_init(void)
 {
 	stm32_pm_context_restore(PM_CTX_DATA, PM_CTX_SIZE);
+
+	/* Clear PM context in BKPSRAM: cold boot at next wake-up */
+	stm32_pm_context_clear();
 }
 
 /*
-- 
2.53.0

